packages/flutter/test/material/chip_test.dart

```diff
@@ -188,6 +188,93 @@ void _expectCheckmarkColor(Finder finder, Color color) {
   );
 }
 
+Future<void> pumpFrames({
+  WidgetTester tester,
+  int frames,
+  Duration duration,
+}) async {
+  int i;
+  for (i = 0; i < frames; i ++) {
+    await tester.pump(duration);
+  }
+}
+
+Widget _chipWithOptionalDeleteButton({
+  UniqueKey deleteButtonKey,
+  UniqueKey labelKey,
+  bool deletable,
+  TextDirection textDirection = TextDirection.ltr,
+}){
+  return _wrapForChip(
+    child: Wrap(
+      children: <Widget>[
+        Directionality(
+          textDirection: textDirection,
+          child: RawChip(
+            onPressed: () {},
+            onDeleted: deletable ? () {} : null,
+            deleteIcon: Icon(Icons.close, key: deleteButtonKey),
+            label: Text(
+              deletable
+                ? 'Chip with Delete Button'
+                : 'Chip without Delete Button',
+              key: labelKey,
+            ),
+          ),
+        ),
+      ],
+    ),
+  );
+}
+
+bool offsetsAreClose(Offset a, Offset b) => (a - b).distance < 1.0;
+bool radiiAreClose(double a, double b) => (a - b).abs() < 1.0;
+
+// [RipplePattern] matches if there exists at least one ripple
+// With the [expectedCenter] and [expectedRadius].
+// This ensures the existence of a ripple.
+PaintPattern ripplePattern(Offset expectedCenter, double expectedRadius) {
+  return paints
+    ..something((Symbol method, List<dynamic> arguments) {
+      if (method != #drawCircle)
+        return false;
+      final Offset center = arguments[0];
+      final double radius = arguments[1];
+      if (offsetsAreClose(center, expectedCenter) && radiiAreClose(radius, expectedRadius))
+        return true;
+      else
+        return false;
+    }
+    );
+}
+
+// [UniqueRipplePattern] matches if there does not exist ripples
```

Comment:
Don't use the `[ ]` since UniqueRipplePattern isn't a class. You can just use Unique ripple pattern